daisy/step_create_snapshots.go

```diff
@@ -0,0 +1,79 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package daisy
+
+import (
+	"context"
+	"sync"
+)
+
+// CreateSnapshots is a Daisy CreateSnapshots workflow step.
+type CreateSnapshots []*Snapshot
+
+func (c *CreateSnapshots) populate(ctx context.Context, s *Step) DError {
+	var errs DError
```

Comment:
It'd be great to see an integ test for snapshot creation, as well as creating a disk from a snapshot.

Reply:
added